app/javascript/react/screens/App/Overview/components/ScheduleMigrationModal/ScheduleMigrationModalBody.js

```diff
@@ -3,54 +3,117 @@ import { Icon } from 'patternfly-react';
 import PropTypes from 'prop-types';
 
 class ScheduleMigrationModalBody extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showDatepicker: false
```

Comment:
This `showDatepicker` state value seems like it's just duplicating the `startMigrationNow` value from the parent component, since you're having to keep them in sync. It might be better to just have `showDatepicker` be a prop, and pass in `showDatepicker={!startMigrationNow}` from the parent so there's a single source of truth for that.

That way, in `componentDidUpdate` you can just detect the prop changing and use that to call the datetimepicker init stuff, and you won't have to call the `startMigrationNowHandler` in there or in `componentDidMount`. You can instead just call that handler in your `handleRadioChange` callback to update the lifted state.

Also `startMigrationNowHandler` is a bit of a confusing name, at first I thought calling it would start the migration. Maybe `setScheduleMode` or something like that would be more clear (I don't feel strongly about that though).

Reply:
I guess you'd still have to make sure you call the datepicker init stuff in componentDidMount if `startMigrationNow` is already false when this mounts, but you can probably do that by moving that stuff to a helper method that you call in both componentDidMount and componentDidUpdate.

Reply:
Yup, fixed too.